www/api/modules/payments/domain/service/subscription_type/subscription_type_service.py

```diff
@@ -26,3 +31,33 @@ def get_all(self):
             subscription_types.append(sub_type._asdict())
 
         return subscription_types
+
+    def create(self,
+               user,
+               subscription_type_data: dict
+               ):
+
+        if user.get("user_type") not in self.__valid_user_types:
+            return "User is not authorized to create a subscription type", None
+
+        error, subscription_type = SubscriptionTypeModel.from_dict(
+            subscription_type_data)
```

Comment:
With the change in the model, sanitization of subscription_type_data is needed, to avoid users defining id's
```suggestion
        error, subscription_type = SubscriptionTypeModel.from_dict(
            subscription_type_data)
```